src/theory/theory_engine.cpp

```diff
@@ -885,9 +910,7 @@ void TheoryEngine::assertToTheory(TNode assertion, TNode originalAssertion, theo
 
   // determine the actual theory that will process/explain the fact, which is
   // THEORY_BUILTIN if the theory uses the central equality engine
-  TheoryId toTheoryIdProp = (Theory::expUsingCentralEqualityEngine(toTheoryId))
-                                ? THEORY_BUILTIN
-                                : toTheoryId;
+  TheoryId toTheoryIdProp = theoryExpPropagation(toTheoryId);
```

Comment:
It seems this changes behaviour?

Reply:
It should be the same I believe?

Reply:
It now considers the `eeMode` option and the result of `usesCentralEqualityEngine()`... if this is intentional, everything is fine.

Reply:
`expUsingCentralEqualityEngine` used tocall to `usesCentralEqualityEngine`, so this was factored out and moved to EqualiyEngineManagerCentral.  I believe the code should be equivalent still.

Reply:
Ah right. Sorry :)